src/ast/CMakeLists.txt

```diff
@@ -66,23 +66,16 @@ if(STATIC_LINKING)
   unlink_transitive_dependency("${CLANG_EXPORTED_TARGETS}" "$<LINK_ONLY:clang-cpp>")
 
   if(TARGET libclang_static)
-    target_link_libraries(ast libclang_static)
+    target_link_libraries(ast PUBLIC libclang_static)
   else()
     # old LLVM versions don't export libclang_static in ClangTargets.cmake; fall back to
     # libclang.a in that case
-    target_link_libraries(ast libclang.a)
+    target_link_libraries(ast PUBLIC libclang.a)
   endif()
 
-  target_link_libraries(ast ${clang_libs})
-  target_link_libraries(ast ${llvm_libs})
+  target_link_libraries(ast PUBLIC ${clang_libs})
+  target_link_libraries(ast PUBLIC ${llvm_libs})
 else()
-  find_library(found_LLVM LLVM HINTS ${LLVM_LIBRARY_DIRS})
-  if(found_LLVM)
-    target_link_libraries(ast LLVM)
-  else()
-    llvm_map_components_to_libnames(_llvm_libs bpfcodegen ipo irreader mcjit orcjit)
-    llvm_expand_dependencies(llvm_libs ${_llvm_libs})
-    target_link_libraries(ast ${llvm_libs})
-  endif()
-  target_link_libraries(ast libclang)
+  llvm_config(ast USE_SHARED bpfcodegen ipo irreader mcjit orcjit)
```

Comment:
> Do I understand correctly that `llvm_config` does the same as the previous code? It would be nice to add a link to the documentation of the macro here or at least a note to the commit message.

Yes, and also, `llvm_config` will do a clever shared library resolution. It's not documented in LLVM documentation, but the macro has been around for about 10 years: https://github.com/llvm/llvm-project/blob/7a8fe0f83c4ba03d07aef9243596d67af74a3b87/llvm/cmake/modules/LLVM-Config.cmake#L72

Reply:
Right, ok. Maybe adding a short comment explaining what the macro does would help here.

Reply:
Will do in a bit

Reply:
Done. Addressed in the latest commit.